test/test-tcp-exportimport.c

```diff
@@ -0,0 +1,240 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "runner.h"
+#include "task.h"
+
+#include <stdio.h>
+#include <string.h>
+
+typedef struct {
+  uv_loop_t* loop;
+  uv_thread_t thread;
+  uv_async_t* recv_channel;
+  uv_async_t* send_channel;
+  uv_tcp_t server;
+  uv_tcp_t conn;
+  int connection_accepted;
+  int close_cb_called;
+} worker_t;
+
+static uv_async_t send_channel;
+static uv_async_t recv_channel;
+static worker_t parent;
+static worker_t child;
+
+static volatile uv_stream_info_t dup_stream;
+
+typedef struct {
+  uv_connect_t conn_req;
+  uv_tcp_t conn;
+} tcp_conn;
+
+#define CONN_COUNT 100
+
+static void close_cb(uv_handle_t* handle) {
+  worker_t* worker = (worker_t*)handle->data;
+  ASSERT(worker);
+  worker->close_cb_called++;
+}
+
+
+static void on_connection(uv_stream_t* server, int status) {
+  int r;
+  worker_t* worker = container_of(server, worker_t, server);
+
+  if (!worker->connection_accepted) {
+    /*
+     * Accept the connection and close it.
+     */
+    ASSERT(status == 0);
+
+    r = uv_tcp_init(server->loop, &worker->conn);
+    ASSERT(r == 0);
+
+    worker->conn.data = worker;
+
+    r = uv_accept(server, (uv_stream_t*)&worker->conn);
+    ASSERT(r == 0);
+
+    worker->connection_accepted = 1;
+
+    uv_close((uv_handle_t*)worker->recv_channel, close_cb);
+    uv_close((uv_handle_t*)&worker->conn, close_cb);
+    uv_close((uv_handle_t*)server, close_cb);
+  }
+}
+
+
+static void close_client_conn_cb(uv_handle_t* handle) {
+  tcp_conn* p = (tcp_conn*)handle->data;
+  free(p);
+}
+
+
+static void connect_cb(uv_connect_t* req, int status) {
+  uv_close((uv_handle_t*)req->handle, close_client_conn_cb);
+}
+
+
+static void make_many_connections(void) {
+  tcp_conn* conn;
+  struct sockaddr_in addr;
+  int r, i;
+
+  for (i = 0; i < CONN_COUNT; i++) {
+    conn = malloc(sizeof(*conn));
+    ASSERT(conn);
+
+    r = uv_tcp_init(uv_default_loop(), &conn->conn);
+    ASSERT(r == 0);
+
+    r = uv_ip4_addr("127.0.0.1", TEST_PORT, &addr);
+    ASSERT(r == 0);
+
+    r = uv_tcp_connect(&conn->conn_req, (uv_tcp_t*)&conn->conn, (struct sockaddr *) &addr, connect_cb);
+    ASSERT(r == 0);
+
+    conn->conn.data = conn;
+  }
+}
+
+
+void on_parent_msg(uv_async_t* handle) {
+  int r;
+
+  ASSERT(dup_stream.type == UV_TCP);
+
+  /* Import the shared TCP server, and start listening on it. */
+  r = uv_tcp_init(parent.loop, &parent.server);
+  ASSERT(r == 0);
+
+  parent.server.data = &parent;
+
+  r = uv_import((uv_stream_t*)&parent.server,
+    (uv_stream_info_t*)&dup_stream);
```

Comment:
```suggestion
               (uv_stream_info_t*)&dup_stream);
```
I hope I got the indent right. :-) It should line up with the first argument, ditto on line 172.